bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/actions/AutomowerActions.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.automower.internal.actions;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.binding.ThingActions;
+import org.eclipse.smarthome.core.thing.binding.ThingActionsScope;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.openhab.binding.automower.internal.things.AutomowerCommand;
+import org.openhab.binding.automower.internal.things.AutomowerHandler;
+import org.openhab.core.automation.annotation.ActionInput;
+import org.openhab.core.automation.annotation.RuleAction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Markus Pfleger - Initial contribution
+ */
+@ThingActionsScope(name = "automower")
+@NonNullByDefault
+public class AutomowerActions implements ThingActions {
+    private final Logger logger = LoggerFactory.getLogger(AutomowerActions.class);
+    private @Nullable AutomowerHandler handler;
+
+    @Override
+    public void setThingHandler(@Nullable ThingHandler handler) {
+        this.handler = (AutomowerHandler) handler;
+    }
+
+    @Override
+    public @Nullable AutomowerHandler getThingHandler() {
+        return handler;
+    }
+
+    @RuleAction(label = "@text/action-start-label", description = "@text/action-start-desc")
+    public void start(
+            @ActionInput(name = "duration", label = "@text/action-input-duration-label", description = "@text/action-input-duration-desc") int durationMin) {
+        AutomowerHandler automowerHandler = handler;
+        if (automowerHandler == null) {
+            logger.warn("Automower Action service ThingHandler is null!");
+        } else {
+            automowerHandler.sendAutomowerCommand(AutomowerCommand.START, durationMin);
+        }
+    }
+
+    public static void start(@Nullable ThingActions actions, int durationMin) {
```

Comment:
Can you add the workaround for the bug discussed in #8116?

Reply:
Hm, not sure. I think this is a really ugly workaround (if you mean the invokeMethodOf stuff)...

Isn't there another solution? Can't I wait for a workaround in the framework and document that the binding suffers from this problem?

Reply:
Since this will only be fixed in OH3, this workaround is getting implemented in all bindings, currently.

Reply:
Ok. Btw. how is the workflow of bringing bindings to openhab 3. Should this be done short before the release? Or asap and then merge all fixes/changes to two branches?

Reply:
Actually OH2 bindings should be compatible with OH3. The hack can be removed when OH3 has been released, I guess.

Reply:
Thx, workaround will be applied in next update to the PR